src/services/bundle_builder.js

```diff
@@ -51,30 +55,39 @@ export default class BundleBuilder {
     };
   }
 
-  async extractBundleDataNecessaryForValidationFromStream(readableStream, bundleVersion) {
+  async validateStreamedBundle(readStream, writeStream, bundleItemsCountLimit) {
+    readStream.on('error', (err) => {
+      writeStream.abort(err);
+    });
+    const [minimalBundleForLatestVersionValidation] = await Promise.all([
+      this.extractBundleDataNecessaryForValidationFromStream(readStream),
+      pipeline(readStream, writeStream)
+    ]);
+    if (!this.supportDeprecatedBundleVersions) {
```

Comment:
This means that even for version 3 bundles, if this config flag is set, then it will always be validated as pre-3. Instead, you should first try to get the version (assume 1 if not pressent) and then only validate against pre-3 if the flag is set.